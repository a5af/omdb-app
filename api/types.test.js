const Film = require('./types').Film;
const fs = require('fs');

const d = fs.readFileSync('mockups.json');
const FILMS = JSON.parse(d);

test("Film's getCrew() returns consistent results", async () => {
  FILMS.forEach((f) => expect(new Film(f).getCrew()).toMatchSnapshot());
});
