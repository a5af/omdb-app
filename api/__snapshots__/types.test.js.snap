// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film's getCrew() returns consistent results 1`] = `
Array [
  "Kenneth Branagh",
  "Michael Green",
  "Agatha Christie",
  "Tom Bateman",
  "Annette Bening",
  "Kenneth Branagh",
]
`;

exports[`Film's getCrew() returns consistent results 2`] = `
Array [
  "Anna Boden",
  "Ryan Fleck",
  "Anna Boden",
  "Ryan Fleck",
  "Geneva Robertson-Dworet",
  "Brie Larson",
  "Samuel L. Jackson",
  "Ben Mendelsohn",
]
`;

exports[`Film's getCrew() returns consistent results 3`] = `
Array [
  "Kenneth Branagh",
  "Ashley Miller",
  "Zack Stentz",
  "Don Payne",
  "Chris Hemsworth",
  "Anthony Hopkins",
  "Natalie Portman",
]
`;

exports[`Film's getCrew() returns consistent results 4`] = `
Array [
  "Quentin Tarantino",
  "Quentin Tarantino",
  "Roger Avary",
  "John Travolta",
  "Uma Thurman",
  "Samuel L. Jackson",
]
`;

exports[`Film's getCrew() returns consistent results 5`] = `
Array [
  "M. Night Shyamalan",
  "M. Night Shyamalan",
  "James McAvoy",
  "Bruce Willis",
  "Samuel L. Jackson",
]
`;
